### DISCUSSION TITLE
IOS Asset Validation failed

### DISCUSSION THREAD

**Discussion by JonasKruegerCode** [codemagic-ci-cd/codemagic-docs#1973] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am building my IOS Flutter app via Codemagic
I have all my App Icons in `Assets.xcassets/AppIcon.appiconset`:
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/e05b8f9c-dc64-4d5a-9826-06c59083d4bf)
Here is an Exerpt from my contents.json file:
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/031e3d93-6eb3-41a7-a2e2-9093ca824833)
Because I had Issues of it detecting specifically the 152 and 167 ones i copied them in to the main Runner-Folder as well and explicitly mentioned them in info.plist:
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/ef2b9b97-5d92-42e5-b635-9ffede5742a7)

### Expected results

The ipa gets published to App Store Connect

### Actual results

The publishing throws the following Error:
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/4950f3f0-b9cd-42e2-b937-2024a2bdb738)

### Build id (optional)

64dcc84aa6df1e9110500271

**Comment by icarusdust** [comment, marked as answer]
Hi @JonasKrueger-techways, have you tried to configure it in Xcode as explained in [this thread](https://medium.com/@this.shoaib/missing-required-icon-file-1893c0bedaaf) ? Check out [this one](https://developer.apple.com/forums/thread/688907) as well.

**Reply by JonasKruegerCode** [reply]
okay the second one helped. I have no access to a mac and thus no access to Xcode. Codemagic now tells me the upload worked
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/129793332/af08f28f-4b59-4ad4-af66-c76656f21699)
but the Build doesn't show up in App Store Connect and thus i cant publish it.

**Reply by JonasKruegerCode** [reply]
okay I just got an Email from Apple with further information. Let me check
